From 3277266d13eb67a5e98dcc943e8a6d567e559afb Mon Sep 17 00:00:00 2001
From: Ren Chenglei <chenglei.ren@intel.com>
Date: Fri, 9 Jun 2023 21:38:39 +0530
Subject: [PATCH] INTERNAL: Fix CtsGraphicsTestCases failures

RENDERING and TEXTURE usage of RGBA_FP16 format is not yet supported.

Tracked-On: OAM-95612
Signed-off-by: Ren Chenglei <chenglei.ren@intel.com>
---
 i915.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/i915.c b/i915.c
index d9da277..2aa04e9 100644
--- a/i915.c
+++ b/i915.c
@@ -227,12 +227,16 @@ static int i915_add_combinations(struct driver *drv)
 	struct i915_device *i915 = drv->priv;
 
 	const uint64_t scanout_and_render = BO_USE_RENDER_MASK | BO_USE_SCANOUT;
-	const uint64_t render = BO_USE_RENDER_MASK;
+	uint64_t render = BO_USE_RENDER_MASK;
 	const uint64_t texture_only = BO_USE_TEXTURE_MASK;
 	// HW protected buffers also need to be scanned out.
 	const uint64_t hw_protected =
 	    i915->has_hw_protection ? (BO_USE_PROTECTED | BO_USE_SCANOUT) : 0;
 
+#ifdef USE_GRALLOC1
+	render = BO_USE_RENDER_MASK & ~(BO_USE_RENDERING | BO_USE_TEXTURE);
+#endif
+
 	const uint64_t linear_mask = BO_USE_RENDERSCRIPT | BO_USE_LINEAR | BO_USE_SW_READ_OFTEN |
 				     BO_USE_SW_WRITE_OFTEN | BO_USE_SW_READ_RARELY |
 				     BO_USE_SW_WRITE_RARELY;
-- 
2.40.1

